Change two optional steps into substeps

adamlazik1 opened the pull request:
In Managing Content guide, section 8.: Changed two optional steps in a procedure to substeps of one optional step. In addition, replaced "Optionally," with "Optional:" in another step for consistency.


* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [x] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

maximiliankolb commented on guides/common/modules/con_managing-content-views.adoc:
```diff
@@ -34,10 +34,11 @@ The general workflow for creating Content Views for filtering and creating snaps
 . Add one or more repositories that you want to the Content View.
-. Optionally, create one or more filters to refine the content of the Content View.
-. Optionally, resolve any package dependencies for a Content View.
+. Optional:
+.. Create one or more filters to refine the content of the Content View.
+.. Resolve any package dependencies for a Content View.
```
```suggestion
. Optional: Create one or more filters to refine the content of the Content View.
. Optional: Resolve any package dependencies for a Content View.
```

adamlazik1 replied:
So merging more optional steps into one is not desired in the documentation I take it?

maximiliankolb replied:
Yes. Also, _we/some_ prefer to not have subtasks within lists.

maximiliankolb commented on guides/common/modules/con_managing-content-views.adoc:
```diff
@@ -34,10 +34,11 @@ The general workflow for creating Content Views for filtering and creating snaps
+.. Resolve any package dependencies for a Content View.
 . Publish the Content View.
-. Optionally, promote the Content View to another environment.
+. Optional: promote the Content View to another environment.
```
```suggestion
. Optional: Promote the Content View to another environment.
```

Levi-Leah commented on guides/common/modules/con_managing-content-views.adoc:
```diff
@@ -34,10 +34,11 @@ The general workflow for creating Content Views for filtering and creating snaps
 . Add one or more repositories that you want to the Content View.
-. Optionally, create one or more filters to refine the content of the Content View.
-. Optionally, resolve any package dependencies for a Content View.
+. Optional:
+.. Create one or more filters to refine the content of the Content View.
+.. Resolve any package dependencies for a Content View.
```
The feedback from https://bugzilla.redhat.com/show_bug.cgi?id=2123123 suggest we make these steps into substeps.
```suggestion
.. Optional: Create one or more filters to refine the content of the Content View.
.. Optional: Resolve any package dependencies for a Content View.
```

Would something like this be feasible?

adamlazik1 replied:
I don't think that we can mark those as sub-steps of the previous step, because previous step describes a separate action.

adamlazik1 commented:
@maximiliankolb Thank you for your clarification. Incorporated the suggested changes.

maximiliankolb reviewed (approved):
LGTM. If you want and if it makes sense, you could also add links to the procedures if there are any.

However, we can also merge as is IMHO. Your call.

adamlazik1 reviewed (commented):
@maximiliankolb I included the links but I am unsure about one of them. I also had one additional idea about improving the section. I left my comments in the review. Could you please take a look and let me know what you think? Thanks!

maximiliankolb commented on guides/common/modules/con_managing-content-views.adoc:
```diff
@@ -34,10 +34,12 @@ The general workflow for creating Content Views for filtering and creating snaps
-. Optionally, create one or more filters to refine the content of the Content View.
-. Optionally, resolve any package dependencies for a Content View.
+. Optional: Create one or more filters to refine the content of the Content View.
+For more information, see xref:Content_Filter_Examples_{context}[].
```
This is fine IMHO.

maximiliankolb commented on guides/common/modules/con_managing-content-views.adoc:
```diff
@@ -34,10 +34,12 @@ The general workflow for creating Content Views for filtering and creating snaps
-. Optionally, resolve any package dependencies for a Content View.
+. Optional: Create one or more filters to refine the content of the Content View.
+For more information, see xref:Content_Filter_Examples_{context}[].
+. Optional: Resolve any package dependencies for a Content View.
```
I tend to add the link here as well, but have no strong feelings.

maximiliankolb reviewed (commented):
LGMT!

Please adjust the commit message; aka. you're not introducing substeps, but merely rewording "optionally".

adamlazik1 commented:
@maximiliankolb Thank you for your review! I made additional changes based on your suggestions:
* Added link for resolving package dependencies into Managing Content Views procedure
* Reworded the commit message.
* Removed the __Package Dependency Resolution__ paragraph. With the link now included in the procedure, it looks to me like it is not needed anymore.

adamlazik1 commented:
Also changed the PR name for consistency.

maximiliankolb reviewed (approved):
LGTM!

maximiliankolb commented:
Merged to master and cherry-picked: ✅ 
   caba9cfda..91a608d43  3.4 -> 3.4
   0e7df03ab..e9cd99c98  3.3 -> 3.3
   f96f63ca7..a6eb81952  3.2 -> 3.2
   f3d2a269d..62d245d98  3.1 -> 3.1
